src/wallet/wallet.cpp

```diff
@@ -3737,7 +3754,20 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
 
         walletInstance->SetBestChain(chainActive.GetLocator());
     }
-    else if (fDisableHotKeys) {
+    else if (IsArgSet("-disablehot")) {
+        bool disableHot = GetBoolArg("-disablehot", DEFAULT_DISABLE_HOT_WALLET);
+        LOCK(walletInstance->cs_wallet);
+        if (walletInstance->vchDefaultKey.IsValid() && disableHot) {
+            InitError(strprintf(_("Error loading %s: You can't disable hot keys if your wallet already contains hot keys"), walletFile));
+            return NULL;
+        }
+        bool hotKeys = walletInstance->IsWalletFlagSet(WALLET_FLAG_DISABLE_HOT_KEYS);
+        if (hotKeys && !disableHot) {
+            InitError(strprintf(_("Error loading %s: You can't enable hot keys once you have initialized a wallet with disabled hot keys"), walletFile));
```

Comment:
> Add `-disablehot` mode: a sane mode for watchonly-wallets

Maybe the error message could suggest passing `-disablehot` to avoid the problem. I think it would make the context of the error easier to understand for someone not expecting it.